pkg/tests/tasks/security/certificate/external_ca_test.go

```diff
@@ -15,109 +15,172 @@
 
-		t.LogStep("Retrieve certificates using the 'openssl s_client -showcerts' command")
-		tmpDir = shell.CreateTempDir(t, "cacerts")
-		oc.Exec(t, pod.MatchingSelector("app=productpage", ns), "istio-proxy",
-			fmt.Sprintf(`openssl s_client -showcerts -connect details:9080 > '%s/bookinfo-proxy-cert.txt' || true`, tmpDir))
-
-		t.LogStep("Extract certificates")
-		shell.Executef(t,
-			`sed -n '/-----BEGIN CERTIFICATE-----/{:start /-----END CERTIFICATE-----/!{N;b start};/.*/p}' '%s/bookinfo-proxy-cert.txt' > '%s/certs.pem'`,
-			tmpDir, tmpDir)
-		shell.Executef(t,
-			`awk 'BEGIN {counter=0;} /BEGIN CERT/{counter++} { print > "%s/proxy-cert-" counter ".pem"}' < '%s/certs.pem'`,
-			tmpDir, tmpDir)
-
-		t.NewSubTest("root certificate").Run(func(t test.TestHelper) {
-			shell.Executef(t, `openssl x509 -in '%s' -text -noout > '%s/root-cert.crt.txt'`, sampleCARoot, tmpDir)
-			shell.Executef(t, `openssl x509 -in '%s/proxy-cert-3.pem' -text -noout > '%s/pod-root-cert.crt.txt'`, tmpDir, tmpDir)
-
-			if err := util.CompareFiles(fmt.Sprintf("%s/root-cert.crt.txt", tmpDir), fmt.Sprintf("%s/pod-root-cert.crt.txt", tmpDir)); err != nil {
-				t.Errorf("Root certs do not match: %v", err)
-			} else {
-				t.LogSuccess("Root certificate received from pod matches the root certificate in cacerts")
-			}
+		var returnedCerts []*x509.Certificate
+		retry.UntilSuccess(t, func(t test.TestHelper) {
+			opensslOutput := oc.Exec(t, pod.MatchingSelector("app=productpage", ns), "istio-proxy",
+				`openssl s_client -showcerts -connect details:9080 || true`) // the "|| true" is needed until oc.Exec can ignore return codes
+			returnedCerts = readPemCertificatesFromText(t, opensslOutput)
 		})
+		certToVerify := returnedCerts[0]
+		otherCerts := returnedCerts[1:]
 
-		t.NewSubTest("CA certificate").Run(func(t test.TestHelper) {
-			shell.Executef(t, `openssl x509 -in '%s' -text -noout > '%s/ca-cert.crt.txt'`, sampleCACert, tmpDir)
-			shell.Executef(t, `openssl x509 -in '%s/proxy-cert-2.pem' -text -noout > '%s/pod-cert-chain-ca.crt.txt'`, tmpDir, tmpDir)
+		verifyContainsCerts(t, otherCerts, chainCerts,
+			"The cert-chain certificates are present in the certificates sent by the tested service",
+			"The cert-chain certificates were not found in the certificates sent by the tested service")
 
-			if err := util.CompareFiles(fmt.Sprintf("%s/ca-cert.crt.txt", tmpDir), fmt.Sprintf("%s/pod-cert-chain-ca.crt.txt", tmpDir)); err != nil {
-				t.Errorf("CA certs do not match: %v", err)
-			} else {
-				t.LogSuccess("CA certificate received from pod matches the CA certificate in cacerts")
+		verifyCertificate(t, certToVerify, []*x509.Certificate{rootCert}, otherCerts,
+			"Certificate trust chain successfully verified",
+			"Unable to verify certificate trust chain")
+	})
+}
+
+func verifyContainsCerts(t test.TestHelper, actualCerts []*x509.Certificate, expectedCerts []*x509.Certificate, successMsg, failureMsg string) {
```

Comment:
nit: could shorten this line a bit:
```suggestion
func verifyContainsCerts(t test.TestHelper, actualCerts, expectedCerts []*x509.Certificate, successMsg, failureMsg string) {
```

Reply:
Good catch! Looks like I'm still shaking off C#